### PR TITLE
add test for model weights equality when fault-tolerant training

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Extends a test by asserting equality of model weights when comparing a full training run with one that got interrupted and resumed. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Dope !

File: tests/utilities/test_auto_restart.py [hunk 1]
```diff
@@ -945,23 +945,28 @@ def test_dataset_rng_states_restart_with_lightning(tmpdir, dataset_classes, mult
...
 
     # the resumed batches should match the batches of the successful training
     all_batches_resumed = torch.stack(complete_batches + resumed_batches)
     assert len(all_batches_resumed) == 9
     assert torch.equal(all_batches, all_batches_resumed)
+
+    # the final weights of a resumed training should equal the weights of an uninterrupted training
+    for w0, w1 in zip(weights0, weights1):
+        assert w0 is not w1
+        assert torch.equal(w0, w1)
```
Comment (ethanwharris) [inline tests/utilities/test_auto_restart.py:972]: Should this maybe use `torch.allclose` instead? Just in case of some floating point weirdness

**Review by ethanwharris — APPROVED** [review]
LGTM, small comment

**Review by justusschock — APPROVED** [review]

File: tests/utilities/test_auto_restart.py (hunk 1 above)
Comment (awaelchli) [reply tests/utilities/test_auto_restart.py]: I'm hesitating, because the whole point of fault-tolerant training is that we can restore exactly the state. While there are some limitations to our implementation, this particular test here is simple and all information to restore deterministically should be given.
